src/org/zalando/stups/friboo/system/new_http.clj

```diff
@@ -0,0 +1,275 @@
+; Copyright © 2016 Zalando SE
+;
+; Licensed under the Apache License, Version 2.0 (the "License");
+; you may not use this file except in compliance with the License.
+; You may obtain a copy of the License at
+;
+;    http://www.apache.org/licenses/LICENSE-2.0
+;
+; Unless required by applicable law or agreed to in writing, software
+; distributed under the License is distributed on an "AS IS" BASIS,
+; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+; See the License for the specific language governing permissions and
+; limitations under the License.
+
+(ns org.zalando.stups.friboo.system.new-http
+  (:require [io.sarnowski.swagger1st.executor :as s1stexec]
+            [io.clj.logging :refer [with-logging-context]]
+            [ring.util.response :as r]
+            [org.zalando.stups.friboo.ring :as ring]
+            [clojure.data.codec.base64 :as b64]
+            [org.zalando.stups.friboo.log :as log]
+            [io.sarnowski.swagger1st.util.api :as api]
+            [io.sarnowski.swagger1st.core :as s1st]
+            [io.sarnowski.swagger1st.util.api :as s1stapi]
+            [ring.adapter.jetty :as jetty]
+            [org.zalando.stups.friboo.security :as security]
+            [ring.middleware.gzip :as gzip])
+  (:import (com.stuartsierra.component Lifecycle)
+           (com.netflix.hystrix.exception HystrixRuntimeException)
+           (org.zalando.stups.txdemarcator Transactions)))
+
+(defn merged-parameters
+  "According to the swagger spec, parameter names are only unique with their type. This one assumes that parameter names
+   are unique in general and flattens them for easier access."
+  [request]
+  (apply merge (vals (:parameters request))))
+
+(defn make-resolver-fn [controller]
+  "Calls operationId function with controller, flattened request params and raw request map."
+  (fn [request-definition]
+    (when-let [operation-fn (s1stexec/operationId-to-function request-definition)]
+      (fn [request]
+        (operation-fn controller (merged-parameters request) request)))))
+
+(defn middleware-chain [context middlewares]
+  (reduce #(%2 %1) context middlewares))
+
+(defn flatten1
+  "Flattens the collection one level, for example, converts {:a 1 :b 2} to (:a 1 :b 2)."
+  [coll]
+  (apply concat coll))
+
+(defn with-flattened-options-map
```

Comment:
Shouldn't you make some of the functions private? This would make a better statement on what the public API for this ns is.